### PR TITLE
add audio track info to event that are sent to MSE

### PR DESCRIPTION
**Pull request by BrandonOCasey**
Tested and ready to review!

### CODE REVIEW

File: lib/mp4/transmuxer.js [hunk 1]
```diff
@@ -862,6 +864,14 @@ CoalesceStream.prototype.flush = function(flushSource) {
     timelineStartPts = this.videoTrack.timelineStartInfo.pts;
   } else if (this.audioTrack) {
     timelineStartPts = this.audioTrack.timelineStartInfo.pts;
+    ['audioobjecttype',
```
Comment (imbcmdth) [inline lib/mp4/transmuxer.js]: I would love it if we could camelCase these names but that would be a more major change so I'm just complaining and don't expect you to change that for this PR

File: lib/mp4/transmuxer.js [hunk 2]
```diff
@@ -862,6 +864,14 @@ CoalesceStream.prototype.flush = function(flushSource) {
...
+     'channelcount',
+     'samplerate',
+     'samplingfrequencyindex',
+     'samplesize'].forEach(function(audioProp) {
+       event.info[audioProp] = self.audioTrack[audioProp];
+     });
```
Comment (imbcmdth) [inline lib/mp4/transmuxer.js]: `forEach` takes a second parameter as the context the function is given when executed.
```
[...].forEach(function(audioProp) {
       event.info[audioProp] = this.audioTrack[audioProp];
     }, this);
```

File: lib/mp4/transmuxer.js [hunk 3]
```diff
@@ -112,11 +130,9 @@ AudioSegmentStream = function(track) {
     collectDtsInfo(track, data);
 
     if (track) {
-      track.audioobjecttype = data.audioobjecttype;
-      track.channelcount = data.channelcount;
-      track.samplerate = data.samplerate;
-      track.samplingfrequencyindex = data.samplingfrequencyindex;
-      track.samplesize = data.samplesize;
+      audioProperties.forEach(function(prop) {
+        track[prop] = data[prop] || '';
```
Comment (gesinger) [inline lib/mp4/transmuxer.js]: Will the empty string default change any of the prior behavior?

**Comment by gesinger** [comment]
One question (as a comment), and do we want to have any tests to check that the properties are in there (besides just checking the info object)? Otherwise, LGTM.

File: lib/mp4/transmuxer.js (hunk 3 above)
Comment (BrandonOCasey) [inline lib/mp4/transmuxer.js]: previously we did not do anything with this information, and from what I have seen it is always defined. If it was not defined the muxer probably would not even be able to get this far. So we could probably forgo the default empty string if we wanted to.

**Comment by BrandonOCasey** [comment]
I will have the tests check that the properties exist

**Comment by gesinger** [comment]
LGTM

File: lib/mp4/transmuxer.js [hunk 4]
```diff
@@ -1145,4 +1164,6 @@ module.exports = {
   Transmuxer: Transmuxer,
   VideoSegmentStream: VideoSegmentStream,
   AudioSegmentStream: AudioSegmentStream,
+  audioProperties: audioProperties,
+  videoProperties: videoProperties
```
Comment (dmlap) [inline lib/mp4/transmuxer.js]: I'm not 100% consistent on it myself but I try to do ALL_CAPS_CASE for constants.

File: lib/mp4/transmuxer.js [hunk 5]
```diff
@@ -822,7 +834,8 @@ CoalesceStream.prototype.flush = function(flushSource) {
     offset = 0,
     event = {
       captions: [],
-      metadata: []
+      metadata: [],
+      info: {}
```
Comment (dmlap) [inline lib/mp4/transmuxer.js]: I kind of wish we had a more descriptive name for this field but it's not worth holding up the PR on.

Comment (dmlap) [inline lib/mp4/transmuxer.js]: Another thought: we need to document these properties at some point.

Comment (BrandonOCasey) [inline lib/mp4/transmuxer.js]: maybe "structure" or "properties"?

**Pull request merged**